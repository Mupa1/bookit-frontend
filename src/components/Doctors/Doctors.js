import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { fetchDoctors } from '../../Redux/actions/index';
import Doctor from '../Doctor/Doctor';
import Sidebar from '../Sidebar/Sidebar';

const Doctors = ({ doctors, setDoctors }) => {
  useEffect(async () => {
    const response = await fetch('https://bookit-doc-appointments-api.herokuapp.com/api/v1/doctors', {
      mode: 'cors',
      headers: {
        Accept: 'application/json, text/plain, */*',
        'Content-Type': 'application/json',
      },
    }).then(res => res.json())
      .catch(error => (error));
    setDoctors(response.data.doctor);
  }, [setDoctors]);

  return (
    <Sidebar content={doctors.length ? (
      <ul>
        {doctors.map(doctor => (
          <Link to={`/doctors/${doctor.id}`} key={doctor.id}>
            <li>
              <Doctor doctor={doctor} />
            </li>
          </Link>
        ))}
      </ul>
    ) : (
      <div className="text-center">Loading doctors...</div>
    )}
    />
  );
};

Doctors.propTypes = {
  doctors: PropTypes.instanceOf(Array).isRequired,
  setDoctors: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  doctors: state.doctors,
});

const mapDispatchToProps = dispatch => ({
  setDoctors: doctors => dispatch(fetchDoctors(doctors)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Doctors);
