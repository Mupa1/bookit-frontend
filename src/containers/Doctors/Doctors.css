.alice-carousel__next-btn {
  right: 0;
  top: 50%;
  border-top-left-radius: 40%;
  border-bottom-left-radius: 40%;
}

.alice-carousel__prev-btn {
  left: 200px;
  top: 50%;
  border-top-right-radius: 40%;
  border-bottom-right-radius: 40%;
}

.alice-carousel__prev-btn,
.alice-carousel__next-btn {
  position: fixed;
  width: 70px;
  background-color: #96b70d;
  padding: 10px;
}

.alice-carousel__next-btn-item,
.alice-carousel__prev-btn-item {
  color: #fff;
  font-weight: bold;
}

.docs-title p {
  color: #a6a6a6;
}

@media screen and (max-width: 768px) {
  .alice-carousel__prev-btn {
    left: 0;
  }
}
