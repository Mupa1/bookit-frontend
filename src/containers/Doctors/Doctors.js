import 'react-alice-carousel/lib/alice-carousel.css';
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import AliceCarousel from 'react-alice-carousel';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { fetchDoctors } from '../../Redux/actions/index';
import Doctor from '../../components/Doctor/Doctor';
import Sidebar from '../../components/Sidebar/Sidebar';
import './Doctors.css';

const Doctors = ({ doctors, setDoctors }) => {
  useEffect(async () => {
    const response = await fetch('http://localhost:3001/api/v1/doctors', {
      mode: 'cors',
      headers: {
        'Content-Type': 'application/json',
      },
    }).then(res => res.json())
      .catch(error => (error));
    setDoctors(response.data.doctor);
  }, [setDoctors]);

  const resp = {
    0: { items: 1 },
    750: { items: 2 },
    1024: { items: 3 },
  };

  return (
    <>
      <Sidebar />
      {doctors.length ? (
        <section className="content">
          <div className="text-center">
            <h2 className="font-weight-bold">OUR DOCTORS</h2>
            <p>Please select a doctor to book an appointment</p>
            <p>...........................</p>
          </div>
          <div>
            <AliceCarousel
              responsive={resp}
              autoPlayInterval={3200}
              autoPlayDirection="ltr"
              autoPlay
              fadeOutAnimation
              mouseTrackingEnabled
              disableAutoPlayOnAction
              dotsDisabled
              infinite
            >
              {doctors.map(doctor => (
                <Link to={`/doctors/${doctor.id}`} key={doctor.id}>
                  <div>
                    <Doctor doctor={doctor} />
                  </div>
                </Link>
              ))}
            </AliceCarousel>
          </div>
        </section>
      ) : (
        <section className="content text-center">
          <h4>Loading doctors...</h4>
        </section>
      )}
    </>
  );
};

Doctors.propTypes = {
  doctors: PropTypes.instanceOf(Array).isRequired,
  setDoctors: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  doctors: state.doctors,
});

const mapDispatchToProps = dispatch => ({
  setDoctors: doctors => dispatch(fetchDoctors(doctors)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Doctors);
