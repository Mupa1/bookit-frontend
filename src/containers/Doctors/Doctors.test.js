import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import Doctors from './Doctors';
import store from '../../Redux/store';

describe('Doctors', () => {
  it('renders Doctors component with a message', () => {
    render(
      <Provider store={store}>
        <Router>
          <Doctors />
        </Router>
      </Provider>,
    );
  });

  it('matches Doctors snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Doctors />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
