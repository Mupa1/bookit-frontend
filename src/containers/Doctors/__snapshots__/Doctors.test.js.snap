// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctors matches Doctors snapshot 1`] = `
Array [
  <button
    className="toggler"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="menu-icon"
      src="https://img.icons8.com/ios-filled/50/000000/menu--v1.png"
    />
  </button>,
  <section
    className="sidebar sidebar"
  >
    <div
      className="header d-flex justify-content-start align-items-center"
    >
      <img
        alt="user"
        className="userlogo"
        src="https://img.icons8.com/carbon-copy/24/000000/user-male-circle.png"
      />
    </div>
     
    <h3
      className="pl-3 font-weight-bold"
    >
      BookIT
    </h3>
    <a
      aria-current={null}
      href="/doctors"
      onClick={[Function]}
    >
      DOCTORS
    </a>
    <a
      aria-current={null}
      href="/appointments"
      onClick={[Function]}
    >
      APPOINTMENTS
    </a>
    <div
      className="footer"
    >
      <div>
        <button
          className="logout"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
       
      <p
        className="d-flex"
      >
        <img
          alt="twitter-icon"
          src="https://img.icons8.com/android/24/000000/twitter.png"
        />
        <img
          alt="fb-icon"
          src="https://img.icons8.com/android/24/000000/facebook-new.png"
        />
        <img
          alt="google-icon"
          src="https://img.icons8.com/android/24/000000/google-plus.png"
        />
        <img
          alt="vimeo-icon"
          src="https://img.icons8.com/material/24/000000/vimeo.png"
        />
        <img
          alt="pinterest-icon"
          src="https://img.icons8.com/metro/24/000000/pinterest.png"
        />
      </p>
    </div>
  </section>,
  <section
    className="content text-center"
  >
    <h4>
      Loading doctors...
    </h4>
  </section>,
]
`;
